tests/checks/indent.fish

```diff
@@ -487,6 +487,54 @@ level 2 } }
     # CHECK: {{^\}$}}
 } | $fish_indent
 
+echo 'false or {
+    echo hi
+}' | $fish_indent
+# CHECK: false or {
+# CHECK: {{^    }}echo hi
+# CHECK: {{^}}}
+
+echo 'true and {
+    echo hi
+}' | $fish_indent
+# CHECK: true and {
+# CHECK: {{^    }}echo hi
+# CHECK: {{^}}}
+
+echo 'test 1 -eq 1; or {
+    echo a
+    echo b
+}' | $fish_indent
+# CHECK: test 1 -eq 1; or {
+# CHECK: {{^    }}echo a
+# CHECK: {{^    }}echo b
+# CHECK: {{^}}}
```

Comment:
we should make the same fix to other constructs that allow the brace statement as "subcommand"
For example:
- if { $multiline_command }
- while { $multiline_command }
- time { $multiline_command }
- not ...
- etc.